Panic when running bootstrap

I am getting a panic when running the bootstrap command on goagen.

Here is the error output:

```
exit status 2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0xa57e8]

goroutine 1 [running]:
panic(0x3cebc0, 0xc4200140b0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/goadesign/goa/design.(*UserTypeDefinition).ToObject(0xc4200ede80, 0xc4202057a8)
        /Users/joshkoch/Projects/go/src/github.com/goadesign/goa/design/types.go:657 +0x28
github.com/goadesign/goa/design/apidsl.Attribute(0x434bd5, 0x2, 0x0, 0x0, 0x0)
        /Users/joshkoch/Projects/go/src/github.com/goadesign/goa/design/apidsl/attribute.go:115 +0x4dc
gitlab.com/koch.josh25/Caster-API/design.glob..func4.1()
        /Users/joshkoch/Projects/go/src/gitlab.com/koch.josh25/Caster-API/design/media_types.go:55 +0x55
github.com/goadesign/goa/dslengine.Execute(0x49d028, 0x64ddc0, 0xc4201b8800, 0x42e0c0)
        /Users/joshkoch/Projects/go/src/github.com/goadesign/goa/dslengine/runner.go:132 +0x8f
github.com/goadesign/goa/design/apidsl.Attributes(0x49d028)
        /Users/joshkoch/Projects/go/src/github.com/goadesign/goa/design/apidsl/media_type.go:318 +0x58
gitlab.com/koch.josh25/Caster-API/design.glob..func4()
        /Users/joshkoch/Projects/go/src/gitlab.com/koch.josh25/Caster-API/design/media_types.go:60 +0x4d
github.com/goadesign/goa/dslengine.Execute(0x49d038, 0x64ddc0, 0xc4201b8800, 0x64ddc0)
        /Users/joshkoch/Projects/go/src/github.com/goadesign/goa/dslengine/runner.go:132 +0x8f
github.com/goadesign/goa/dslengine.runSet(0xc420206ae0, 0x3, 0x3, 0x0, 0x3)
        /Users/joshkoch/Projects/go/src/github.com/goadesign/goa/dslengine/runner.go:297 +0x25e
github.com/goadesign/goa/design.(*APIDefinition).IterateSets(0xc420162dc0, 0x49ce40)
        /Users/joshkoch/Projects/go/src/github.com/goadesign/goa/design/definitions.go:479 +0x2d6
github.com/goadesign/goa/dslengine.Run(0x64f6c0, 0xc42020c080)
        /Users/joshkoch/Projects/go/src/github.com/goadesign/goa/dslengine/runner.go:97 +0x103
main.main()
        /Users/joshkoch/Projects/go/src/gitlab.com/koch.josh25/Caster-API/goagen915648630/main.go:26 +0x63
```

Here is the media type file that seems to be causing the problem:

```
var Account = MediaType("application/vnd.account+json", func() {
	Description("Account")
	Attributes(func() {
		Attribute("id", Integer, "ID of account")
		Attribute("email", String, "User email")
		Attribute("podcasts", CollectionOf(PodcastMedia), "User podcasts")
		Attribute("created_at", DateTime, "Date of creation")
		Attribute("updated_at", DateTime, "Date of last update")

		Required("id", "email", "created_at")
	})

	View("default", func() {
		Attribute("id")
		Attribute("email")
		Attribute("podcasts")
		Attribute("created_at")
	})
	View("tiny", func() {
		Attribute("id")
		Attribute("email")
	})
})

var PodcastMedia = MediaType("application/vnd.podcast+json", func() {
	Reference(Podcast)
	Attributes(func() {
		Attribute("id")
		Attribute("title")
		Attribute("author")
		Attribute("description")
		Attribute("url")
		Attribute("episodes", CollectionOf(EpisodeMedia), "Podcast episodes")
	})
	View("default", func() {
		Attribute("id")
		Attribute("title")
		Attribute("author")
		Attribute("description")
		Attribute("url")
		Attribute("episodes")
	})
})

var EpisodeMedia = MediaType("application/vnd.episode+json", func() {
	Reference(Episode)
	Attributes(func() {
		Attribute("id")
		Attribute("title")
		Attribute("description")
		Attribute("url")
		Attribute("length")
	})
	View("default", func() {
		Attribute("id")
		Attribute("title")
		Attribute("description")
		Attribute("url")
		Attribute("length")
	})
})
```

Here are the type definitions for the media types: 

```
var Podcast = Type("Podcast", func() {
	Description("Podcast")
	Attributes(func() {
		Attribute("id", Integer, "Podcast id")
		Attribute("title", String, "Podcast title")
		Attribute("author", String, "Podcast author")
		Attribute("description", String, "Podcast decription")
		Attribute("url", String, "Podcast url")
	})
})

var Episode = Type("Episode", func() {
	Description("Episode")
	Attributes(func() {
		Attribute("id", Integer, "Episode id")
		Attribute("title", String, "Episode title")
		Attribute("description", String, "Episode decription")
		Attribute("url", String, "Episode url")
		Attribute("length", Number, "Episode length")
	})
})
```

--- a/design/apidsl/attribute.go
+++ b/design/apidsl/attribute.go
@@ -111,7 +111,7 @@ func Attribute(name string, args ...interface{}) {
 		}
 
 		var baseAttr *design.AttributeDefinition
-		if parent.Reference != nil {
+		if parent.Reference != nil && parent.Reference.IsObject() {
 			if att, ok := parent.Reference.ToObject()[name]; ok {
 				baseAttr = design.DupAtt(att)
 			}

--- a/design/types.go
+++ b/design/types.go
@@ -639,19 +639,19 @@ func (u *UserTypeDefinition) Kind() Kind { return UserTypeKind }
 func (u *UserTypeDefinition) Name() string { return u.Type.Name() }
 
 // IsPrimitive calls IsPrimitive on the user type underlying data type.
-func (u *UserTypeDefinition) IsPrimitive() bool { return u.Type.IsPrimitive() }
+func (u *UserTypeDefinition) IsPrimitive() bool { return u.Type != nil && u.Type.IsPrimitive() }
 
 // HasAttributes calls the HasAttributes on the user type underlying data type.
 func (u *UserTypeDefinition) HasAttributes() bool { return u.Type.HasAttributes() }
 
 // IsObject calls IsObject on the user type underlying data type.
-func (u *UserTypeDefinition) IsObject() bool { return u.Type.IsObject() }
+func (u *UserTypeDefinition) IsObject() bool { return u.Type != nil && u.Type.IsObject() }
 
 // IsArray calls IsArray on the user type underlying data type.
-func (u *UserTypeDefinition) IsArray() bool { return u.Type.IsArray() }
+func (u *UserTypeDefinition) IsArray() bool { return u.Type != nil && u.Type.IsArray() }
 
 // IsHash calls IsHash on the user type underlying data type.
-func (u *UserTypeDefinition) IsHash() bool { return u.Type.IsHash() }
+func (u *UserTypeDefinition) IsHash() bool { return u.Type != nil && u.Type.IsHash() }
 
 // ToObject calls ToObject on the user type underlying data type.
 func (u *UserTypeDefinition) ToObject() Object { return u.Type.ToObject() }

--- a/design/types_test.go
+++ b/design/types_test.go
@@ -12,6 +12,68 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+var _ = Describe("IsObject", func() {
+	var dt DataType
+	var isObject bool
+
+	JustBeforeEach(func() {
+		isObject = dt.IsObject()
+	})
+
+	Context("with a primitive", func() {
+		BeforeEach(func() {
+			dt = String
+		})
+
+		It("returns false", func() {
+			Ω(isObject).Should(BeFalse())
+		})
+	})
+
+	Context("with an array", func() {
+		BeforeEach(func() {
+			dt = &Array{ElemType: &AttributeDefinition{Type: String}}
+		})
+
+		It("returns false", func() {
+			Ω(isObject).Should(BeFalse())
+		})
+	})
+
+	Context("with a hash", func() {
+		BeforeEach(func() {
+			dt = &Hash{
+				KeyType:  &AttributeDefinition{Type: String},
+				ElemType: &AttributeDefinition{Type: String},
+			}
+		})
+
+		It("returns false", func() {
+			Ω(isObject).Should(BeFalse())
+		})
+	})
+
+	Context("with a nil user type type", func() {
+		BeforeEach(func() {
+			dt = &UserTypeDefinition{AttributeDefinition: &AttributeDefinition{Type: nil}}
+		})
+
+		It("returns false", func() {
+			Ω(isObject).Should(BeFalse())
+		})
+	})
+
+	Context("with an object", func() {
+		BeforeEach(func() {
+			dt = Object{}
+		})
+
+		It("returns true", func() {
+			Ω(isObject).Should(BeTrue())
+		})
+	})
+})
+
 var _ = Describe("Project", func() {
 	var mt *MediaTypeDefinition
 	var view string

